editing/edit-feature-attachments/src/main/java/com/esri/samples/edit_feature_attachments/EditFeatureAttachmentsSample.java

```diff
@@ -160,32 +155,31 @@ public void start(Stage stage) {
           attachmentList.getItems().clear();
 
           // get the clicked feature
-          ListenableFuture<IdentifyLayerResult> results = mapView.identifyLayerAsync(featureLayer, point, 1, false);
-          results.addDoneListener(() -> {
-            try {
-              IdentifyLayerResult layer = results.get();
-              List<GeoElement> identified = layer.getElements();
-              if (identified.size() > 0) {
-                GeoElement element = identified.get(0);
-                // get selected feature
-                if (element instanceof ArcGISFeature) {
-                  selected = (ArcGISFeature) element;
-                  featureLayer.selectFeature(selected);
-                  selected.loadAsync();
-                  selected.addDoneLoadingListener(() -> {
-                    if (selected.getLoadStatus() == LoadStatus.LOADED) {
-                      fetchAttachments(selected);
-                    } else {
-                      displayMessage("Error", "Element failed to load!");
-                    }
-                  });
-                  addAttachmentButton.setDisable(false);
+          mapView.identifyLayerAsync(featureLayer, point, 1, false).toCompletableFuture()
+            .whenComplete((layer, ex) -> {
+              if (ex == null) {
+                List<GeoElement> identified = layer.getElements();
```

Comment:
I think we can add a comment within like
```suggestion
               // if the operation completes successfully, get the result
                List<GeoElement> identified = layer.getElements();
```

Reply:
I won't add this in each sample, but I think it's good doc to add where relevant across them all

Reply:
Not sure this is quite right - the layer _is_ the result. Not like before where we'd have a `IdentifyLayerResult` (or similar) which we'd need to call `get().getElements()` on (and thus surround with try/catch). Would something like "if the identification operation completes successfully, get the list of identified geoElements" be better?

Reply:
Happy for you to devise the comments as appropriate yep, it was more to demonstrate that a comment either side of the added if/else would be good and easier to pick them out.